sdk/go/pulumi/context_test.go

```diff
@@ -561,3 +561,45 @@ func (c *resmonClientWithFeatures) SupportsFeature(
 	}
 	return c.ResourceMonitorClient.SupportsFeature(ctx, req, opts...)
 }
+
+func TestSourcePosition(t *testing.T) {
+	t.Parallel()
+
+	mocks := &testMonitor{
+		NewResourceF: func(args MockResourceArgs) (string, resource.PropertyMap, error) {
+			var sourcePosition *pulumirpc.SourcePosition
+			switch {
+			case args.RegisterRPC != nil:
+				sourcePosition = args.RegisterRPC.SourcePosition
+			case args.ReadRPC != nil:
+				sourcePosition = args.ReadRPC.SourcePosition
+			}
+
+			require.NotNil(t, sourcePosition)
+			assert.True(t, strings.HasSuffix(sourcePosition.Uri, "context_test.go"))
```

Comment:
I guess it's too brittle to check the lines here, if someone adds new lines above this function the test would break.

Reply:
Yeah, that was my thought as well.